core/src/main/java/org/apache/iceberg/io/RollingFileWriter.java

```diff
@@ -109,7 +109,7 @@ protected void openCurrentWriter() {
   }
 
   private EncryptedOutputFile newFile() {
-    if (partition == null) {
+    if (spec.isUnpartitioned() || partition == null) {
```

Comment:
this check needs to be added to other writers, like `FanoutWriter`, `ClusteredWriter`. They all have the same pattern. 
```
    return partition == null
        ? fileFactory.newOutputFile()
        : fileFactory.newOutputFile(spec, partition);
  }
```

Originally I was wondering if this check should be done in the method below of `OutputFileFactory` class. But I see it is not appropriate, as the Java doc clearly indicates that this method should be used for partitioned writes. Hence caller needs to separate out unpartitioned vs partitioned writes.
```
  /** Generates an {@link EncryptedOutputFile} for partitioned writes in a given spec. */
  public EncryptedOutputFile newOutputFile(PartitionSpec spec, StructLike partition) 
```

Reply:
I'm fine with adding those as well. WDYT @aokolnychyi